### PR TITLE
refactor(error): dedicated error type for `dml` crate

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Introduce dedicated error type `DmlError` for `dml` crate, to clean-up the usage of `anyhow` and `RwError`. Also change the signature of `StreamReaderWithPause` to accept any streams that yields `Err = StreamExecutorError` instead of `RwError`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/stream_reader.rs [hunk 1]
```diff
@@ -48,26 +46,14 @@ pub(super) struct StreamReaderWithPause<const BIASED: bool, M> {
 }
 
 impl<const BIASED: bool, M: Send + 'static> StreamReaderWithPause<BIASED, M> {
-    /// Receive messages from the reader. Hang up on error.
-    #[try_stream(ok = M, error = StreamExecutorError)]
-    async fn data_stream(stream: BoxTryStream<M>) {
```
Comment (BugenZhao) [inline src/stream/src/executor/stream_reader.rs:53]: This looks like a no-op for `FusedStream`s. So I removed it.

**Review by chenzl25 — APPROVED** [review]
LGTM

File: src/stream/src/executor/stream_reader.rs (hunk 1 above)
Comment (chenzl25) [reply src/stream/src/executor/stream_reader.rs:53]: I also don't know why we need to do so before.

**Review by stdrc — APPROVED** [review]
approve on behalf of cargo.lock

**Pull request merged**